src/arche/tools/s3.py

```diff
@@ -42,5 +42,16 @@ def get_contents_as_string(bucket: str, filepath: str) -> str:
 
 
 def get_contents(url: str) -> str:
+    """Fetch file contents from the remote source. Only s3, https schemes are allowed
+    """
+    o = urlparse(url)
+    netloc = o.netloc
+    relative_path = o.path.lstrip("/")
+    if not netloc or not relative_path:
+        raise ValueError(f"'{url}' is not a valid s3 or URL path to a file")
+    if o.scheme not in ["s3", "https"]:
```

Comment:
:lipstick: Instead of:

```
if scheme not in ('a', 'b'):
    raise ValueError()
if scheme == 'a':
    return a()
return b()
```

It would look a bit cleaner to me to go with:

```
if scheme == 'a':
    return a()
if scheme == 'b':
    return b()
raise ValueError()
```